1.1/geo.ttl

```diff
@@ -540,7 +568,8 @@ geosparql:
 	rdfs:subClassOf :SpatialObject ;
 	owl:disjointWith :Geometry ;
 	rdfs:isDefinedBy geosparql: , geosparql-spec: ;
-	skos:definition """This class represents the top-level feature type. This class is equivalent to GFI_Feature defined in ISO 19156:2011, and it is superclass of all feature types."""@en ;
+	skos:definition """A spatial phenomenon in a universe of discourse."""@en ;
+	skos:note """A Feature can be discrete (having a well-defined boundary or extent, for example an apple or a river) or can be continuous (for example: temperature, elevation)."""@en ;
```

Comment:
@dr-shorthair can you please consider this definition?

Reply:
Yes, this definition contains some seeds of confusion. 

There are generally two views of geospatial information. The simplified version is 
- discrete objects recognised in space and time, each of which is classified with a type, and has a set of properties that are characteristic of that type, each property (height, weight, location, colour) being constant or fixed in the object
- properties that vary continuously, though not necessarily smoothly, across space-time in a 'field' - temperature, pressure, density, material, ownership etc. 

In GIS these views essentially correspond to the 'vector' and 'raster' or gridded representation. 
In the GIS standards space (OGC etc) the objects in the vector representation are known as 'features', while the fields are known as 'coverages'. 

This is a simplified taxonomy, because of course there are objects or features that have properties that vary within them; and features or objects can be detected within fields or coverages (@kjano 's storm front)
It is also somewhat troublesome as the GIS term **feature** does not really match the natural language semantics so is confusing to non GIS people, and **coverage** is used with a different meaning in some nearby knowledge representations - e.g. Dublin Core where it means 'extent'. 
And there are philosophers who will argue for grey boundaries between these meta-models on other grounds. 
But it has proved to be useful in practice and these are the terms used in GIS standards. 

In this context, the `note` above is troublesome. 
> skos:note "A Feature can be discrete (having a well-defined boundary or extent, for example an apple or a river) or can be continuous (for example: temperature, elevation)."@en ;

The first half seems to refer to 'features' as I just described, while the second half appears to refer to 'coverages', which are usually modelled in a very different way. 
IMAO the top level class hierarchy in GeoSPARQL really only addresses the object/feature view, and does not effectively touch the field/coverage view. That is a scope limitation that can't and probably shouldn't be covered over with a note alongside the key definition.

Reply:
I agree with this view. Unless we have properly integrated coverages in GeoSPARQL I would change this note to only discrete features for now.

Reply:
The common/official definition of _coverage_ seems to be "feature that acts as a function to return values from its range for any direct position within its spatial, temporal or spatiotemporal domain" (taken from [OpenGISAbstract Specification Topic 6: Schema for coverage geometry and functions](https://portal.ogc.org/files/?artifact_id=19820)). So: a coverage class would be a subclass of `Feature`.
Of course GeoSPARQL could go against that. But at this moment we can not exclude coverages ever being included in the GeoSPARQL ontology.  I have included mentioning continuous phenomena to pave the way for extending the ontology with coverages (by the WG or other parties). 
@dr-shorthair: is there any semantic clash at the moment if features in GeoSPARQL are meant to include coverages?
@situx : if we only mention discrete features in the note, that would almost actively exclude continuous features. And that would block or impede extending GeoSPARQL to coverages in the future.

Reply:
I agree with @dr-shorthair nice overview. These terms (as in any other domain as well) are not perfect, but they are used in GIS, GIScience, and related textbooks for decades. How to deal with coverages is still an ongoing discussion in the geo-semantics/RDF realm (DGGS may be one approach). Hence I would assume 'Feature' to really stand for 'A representation of a real-world object on a map.' (https://support.esri.com/en/other-resources/gis-dictionary/term/dcc335be-78ae-4bd2-b254-b44c37343f75). The point here is not so much the 'map' nor the 'real-world' but the so-called 'Object' view (as compared to 'Field').

What about the following:

`	skos:note """A Feature is a representation of a uniquely identifiable object in the physical world together with its properties and relationships to other features. While all objects (and therefore the features used to represent them) are bounded, their boundaries may be crisp (e.g., the declared boundaries of a state) or vague (e.g., the delineation of a valley versus its neighboring mountain). While discrete in nature, features may be created from continuous observations, e.g., coverages, such as an isochrone that determines the region that can be reached by ambulance within 5 minutes. """@en ;`

This is also inspired by Kuhn's and Goodchild's work on the topic. In general, I feel that a bit more text often helps to avoid confusion by readers from the border RDF/SW/LD/KG community.

Reply:
> What about the following:
> 
> ` skos:note """A Feature is a representation of a uniquely identifiable object in the physical world together with its properties and relationships to other features. While all objects (and therefore the features used to represent them) are bounded, their boundaries may be crisp (e.g., the declared boundaries of a state) or vague (e.g., the delineation of a valley versus its neighboring mountains). While discrete in nature, features may be created from continuous observations, e.g., coverages, such as an isochrone that determines the region that can be reached by ambulance within 5 minutes. """@en ;`

@kjano, I agree with you assessment that more text increases understanding. Even within the GIS community it is easy to have confusion about what is meant exactly. 
Two things catch my attention:

1. The proposed note introduces the idea of the physical world. I hope we can work without that limitation. Firstly, it is easy to come up with borderline cases in which it is not clear whether something is an object in the physical world. How about historical places that have not been discovered yet and may have never existed? How about predictions for the future that haven't materialized yet? Secondly, this scope precludes use of GeoSPARQL for virtual worlds that are used in e.g. (serious) gaming. I think that would be a pity.
2. Scoping `Feature` to discrete spatial objects is a choice we can make. Coverages could later be subclassed from `SpatialObject`. But since the term _feature_ is used extensively by the OGC, shouldn't we make it very clear that the GeoSPARQL definition is not the same as the OGC definition?

Reply:
Thanks a lot for your reply @FransKnibbe. OGC uses the same definition as I. See here http://docs.ogc.org/is/17-069r3/17-069r3.html#_terms_and_definitions and the Simple Feature Specs where 'Feature' is defined as 'abstraction of real world phenomena [ISO 19101-1:2014]'. I agree that the term 'real' is problematic and thus changed it to physical. I also understand that this is equally not perfect, so let us just drop any 'reals' and 'perfects' and simply go with 'world'. I also switch to the term 'phenomenon' to keep in line with the OGC language.

`skos:note """A Feature represents a uniquely identifiable phenomenon in the physical world together with its properties and relationships to other features. While such phenomena (and therefore the features used to represent them) are bounded, their boundaries may be crisp (e.g., the declared boundaries of a state), vague (e.g., the delineation of a valley versus its neighboring mountains), and change with time (e.g., a storm front). While discrete in nature, features may be created from continuous observations, e.g., coverages, such as an isochrone that determines the region that can be reached by ambulance within 5 minutes. """@en ;`

Reply:
@kjano, in the definition of `Feature` (we are talking about the `skos:note` now) I tried to have a more open designation as an alternative to _real_ or _physical world_: universe of discourse. It is a term I found in an OGC specification somewhere. I thought it would be useful to open up the `Feature` class and its related ontology terms to virtual worlds too (and other celestial bodies, should anyone read 'world' as a synonym for 'Earth').

As for a conflict with the OGC definition: it seems that OGC specifications explicitly model continuous, field-like phenomena as coverages, which are a type of feature. And the proposed note for `geo:Feature` seems to explicitly target discrete spatial objects. So I do sense a contradiction between GeoSPARQL and other OGC specifications in that case (to complicate matters, [the coverage specification](https://portal.ogc.org/files/?artifact_id=19820) distinguishes discrete and continuous coverages). I am not for or against modelling coverages as features, but I do think we need to be very clear in what we mean.

Reply:
> coverages, which are a type of feature 

In principle, yes. In practice, not so much. WCS is not a specialization of WFS, for example. 

'Phenomenon' is also problematic. It is at least as general or maybe more so than 'feature'. 

I would suggest explicitly limiting GeoSPARQL to discrete phenomena/features since that is where it is most useful (for now at least). If this conflicts with some of the more esoteric understandings from OGC, ISO, then that doesn't worry me too much because in practice these have little effect on implementations and interfaces. 

In the [OGC Reference Model](https://portal.ogc.org/files/?artifact_id=47245) discusses the issue in Section 2.5: 

> Geographic phenomena fall into two broad categories, discrete and continuous. Discrete
phenomena are recognizable objects that have relatively well-defined boundaries or
spatial extent. Examples include buildings, streams, and measurement stations.
Continuous phenomena vary over space and have no specific extent. Examples include
temperature, soil composition, and elevation. A value or description of a continuous 
phenomenon is only meaningful at a particular position in space (and possibly time).
Temperature, for example, takes on specific values only at defined locations, whether
measured or interpolated from other locations. These concepts are not mutually
exclusive. In fact, many components of the landscape may be viewed alternatively as
discrete or continuous. Historically, geographic information has been treated in terms of
two fundamental types called vector data and raster data.

ISO 19123 has the formal definition of 'Coverage' 

> feature that acts as a function to return values from its range for any direct position within its spatial, temporal or spatiotemporal domain

So yes, a Coverage is a Feature, but not a discrete one.

Reply:
> So yes, a Coverage is a Feature, but not a discrete one.

Thank you for the guidance, @dr-shorthair. I will try to think up a more narrow definition of `Feature`. I was already uncomfortable with there being very little to distinguish `SpatialObject` from `Feature`. And yes, 'phenomenon' is just another way of saying 'thing' or 'object'; it doesn't do much in the way of demarcation. 

For some extra clarity: would you recommend explicitly excluding coverages from the definition of `Feature`? Because discrete things can be modelled as coverages too...
The  [Abstract Specification Topic 6: Schema for coverage geometry and functions](https://portal.ogc.org/files/?artifact_id=19820) expands the text from section 2.5 of the OGC reference model in the following way:

> In this International Standard, coverage is a subtype of feature. A coverage is a feature that has multiple values for each attribute type, where each direct position within the geometric representation of the feature has a single value for each attribute type. Just as the concepts of discrete and continuous phenomena are not mutually exclusive, their representations as discrete features or coverages are not mutually exclusive. The same phenomenon may be represented as either a discrete feature or a coverage. A city may be viewed as a discrete feature that returns a single value for each attribute, such as its name, area and total population. The city feature may also be represented as a coverage that returns values such as population density, land value or air quality index for each position in the city.

To me it seems most logical to separate real-world things from the way they can be modelled or represented in data structures. So coverage would be a way to model or codify a feature, similar to geometry. Should we model in UML, they could be composite datatypes. That would also mean that if GeoSPARQL was to be extended to support coverages, coverage would be modeled similar to `Geometry`, i.e. as a subclass of `SpatialObject`, with `Feature` having a property `hasCoverage`. If this is agreed, I think it would be good to update the note on `Geometry` too, to explain that it is a way to describe a `Feature` as structured data. From this perspective, the definition of `Feature` does not have to say anything about coverages.